napari_btrack/track.py

```diff
@@ -1,24 +1,144 @@
+# widgets for which the default widget type is incorrect
+hidden_variable_names = [
+    "name",
+    "measurements",
+    "states",
+    "dt",
+    "apoptosis_rate",
+    "prob_not_assign",
+    "eta",
+]
+all_hypotheses = ["P_FP", "P_init", "P_term", "P_link", "P_branch", "P_dead"]
+
+
+@dataclass
+class Matrices:
+    """A helper dataclass to adapt matrix representation to and from pydantic.
+    This is needed because TrackerConfig stored "scaled" matrices, i.e.
+    doesn't store sigma and the "unscaled" matrix separately.
+    """
+
+    names: List[str] = field(default_factory=lambda: ["A", "H", "P", "G", "R"])
+    default_sigmas: List[float] = field(
+        default_factory=lambda: [1.0, 1.0, 150.0, 15.0, 5.0]
+    )
+    unscaled_matrices: Dict[str, List[float]] = field(
+        default_factory=lambda: dict(
+            # stop black from reformatting here
+            # so we can read the matrix entries
+            # fmt: off
+            A_cell=[1, 0, 0, 1, 0, 0,
+                    0, 1, 0, 0, 1, 0,
+                    0, 0, 1, 0, 0, 1,
+                    0, 0, 0, 1, 0, 0,
+                    0, 0, 0, 0, 1, 0,
+                    0, 0, 0, 0, 0, 1
+                    ],
+            A_particle=[1, 0, 0, 0, 0, 0,
+                        0, 1, 0, 0, 0, 0,
+                        0, 0, 1, 0, 0, 0,
+                        0, 0, 0, 1, 0, 0,
+                        0, 0, 0, 0, 1, 0,
+                        0, 0, 0, 0, 0, 1],
+            H=[1, 0, 0, 0, 0, 0,
+               0, 1, 0, 0, 0, 0,
+               0, 0, 1, 0, 0, 0],
+            P=[0.1, 0, 0, 0, 0, 0,
+                0, 0.1, 0, 0, 0, 0,
+                0, 0, 0.1, 0, 0, 0,
+                0, 0, 0, 1, 0, 0,
+                0, 0, 0, 0, 1, 0,
+                0, 0, 0, 0, 0, 1],
+            G=[0.5, 0.5, 0.5, 1, 1, 1],
+            R=[1, 0, 0,
+                0, 1, 0,
+                0, 0, 1],
+            # fmt: on
+        )
+    )
+
+    @classmethod
+    def get_scaled_matrix(
+        cls, name: str, sigma: float, use_cell_config: bool = True
+    ) -> List[float]:
+        """Returns the scaled version (i.e. the unscaled matrix multiplied by sigma)
+        of the matrix.
+
+        Keyword arguments:
+        name -- the matrix name (can be one of A, H, P, G, R)
+        sigma -- the factor to scale the matrix entries with
+        cell -- whether to use cell config matrices or not (default true)
+        """
+        if name == "A":
+            if use_cell_config:
+                name = "A_cell"
+            else:
+                name = "A_particle"
+        return (np.asarray(cls().unscaled_matrices[name]) * sigma).tolist()
+
+    @classmethod
+    def get_sigma(cls, name: str, scaled_matrix: npt.NDArray[np.float64]) -> float:
+        """Returns the factor sigma which is the multiplier between the given scaled
+        matrix and the unscaled matrix of the given name.
+
+        Note: Calculation relies on the top-left entry of the matrix.
```

Comment:
```suggestion
        Note: The calculation is done with the top-left entry of the matrix, and all other entries are ignored.
```
Something like this is more precise.